src/backend/distributed/worker/worker_merge_protocol.c

```diff
@@ -221,11 +223,17 @@ worker_merge_files_and_run_query(PG_FUNCTION_ARGS)
 							   createMergeTableQuery)));
 	}
 
+	/* need superuser to copy from files */
```

Comment:
@marcocitus can you have a look at this? I ran into issues running this udf as non-root and it differed here compared to the call to `CopyTaskFilesFromDirectory` in [`worker_merge_files_into_table`](https://github.com/citusdata/citus/blob/master/src/backend/distributed/worker/worker_merge_protocol.c#L130-L137)